### PR TITLE
Feature/chat

### LINKED ISSUE #7: feat: patients should be able to chat with consultants

### PR DESCRIPTION
**Pull request by lovisgod**
## Description
This PR implements chatting between patients and consultants.
Socket IO was used.

All pull requests should be related to open issues. Indicate the issue(s) bellow and remove this line:
Fixes #7 

## How Has This Been Tested?
1.  Clone/Pull/Fetch this branch.
2.  Run `npm install` to install the dependencies.
3.  Run `npm test` to run the tests.
4.  Log in and get a token from the system.
5.  Use [FireCamp](https://chrome.google.com/webstore/detail/firecamp-a-campsite-for-d/eajaahbjpnhghjcdaclbkeamlkepinbl) to test the socket connection.

## Screenshots (if applicable, else remove this line / section)
 
sockettest

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented on my code, particularly in hard-to-understand areas
-   [x] I have added necessary inline code documentation
-   [x] I have added tests that prove my fix is effective and that this feature works
-   [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/database/migrations/20200516230350-create-chat-table.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    user_uuid: {
+      allowNull: false,
+      type: Sequelize.UUID
+    },
+    connection_uuid: {
+      allowNull: false,
+      type: Sequelize.UUID
+    },
+    parent_uuid: {
```
Comment (adaezeodurukwe) [inline src/database/migrations/20200516230350-create-chat-table.js:18]: parent uuid? what does this identify?

File: src/middlewares/SocketAuth.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const socketAuth = async (token, socket, io) => {
+  try {
+    if (!token) io.to(socket.id).emit('login_error', { message: 'Access denied, please login to have access' });
+    const { email } = verifyToken(token);
+    const user = await User.findOne({
+      where: { email },
+      attributes: {
+        exclude: ['password']
+      }
+      // include: ['token'],
```
Comment (adaezeodurukwe) [inline src/middlewares/SocketAuth.js:15]: remove commented code

File: src/models/Chat.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    'Chat',
+    {
+      uuid: {
+        type: DataTypes.UUID,
+        defaultValue: DataTypes.UUIDV4,
+        primaryKey: true
+      },
+      user_uuid: DataTypes.UUID,
+      connection_uuid: DataTypes.UUID,
+      parent_uuid: DataTypes.UUID,
```
Comment (adaezeodurukwe) [inline src/models/Chat.js:12]: parent uuid should be checked

File: src/database/migrations/20200516230350-create-chat-table.js (hunk 1 above)
Comment (John-Akins) [reply src/database/migrations/20200516230350-create-chat-table.js:18]: That's the sender id for that particular message

Comment (lovisgod) [reply src/database/migrations/20200516230350-create-chat-table.js:18]: Lets's say a message is a response to another message, then the parent_uuid is the uuid  of the former message that this present message is a response to.

File: src/models/Chat.js (hunk 3 above)
Comment (lovisgod) [reply src/models/Chat.js:12]: how?

File: src/middlewares/SocketAuth.js (hunk 2 above)
Comment (lovisgod) [reply src/middlewares/SocketAuth.js:15]: will do.

**Review by adaezeodurukwe — APPROVED** [review]

**Pull request merged**